source/boundary_composition/spherical_constant.cc

```diff
@@ -42,24 +43,23 @@ namespace aspect
     {
       (void)geometry_model;
 
-      // verify that the geometry is in fact a spherical shell since only
-      // for this geometry do we know for sure what boundary indicators it
-      // uses and what they mean
-      Assert (dynamic_cast<const GeometryModel::SphericalShell<dim>*>(&geometry_model)
-              != 0,
-              ExcMessage ("This boundary model is only implemented if the geometry is "
-                          "in fact a spherical shell."));
+      // verify that the geometry is a spherical shell or a chunk since only
+      // for geometries based on spherical shells do we know which boundary indicators are
+      // used and what they mean
+      Assert ((dynamic_cast<const GeometryModel::SphericalShell<dim>*>(&geometry_model) != 0
+               || dynamic_cast<const GeometryModel::Chunk<dim>*>(&geometry_model) != 0),
+              ExcMessage ("This boundary model is only implemented if the geometry "
+                          "is a spherical shell or chunk."));
 
-      switch (boundary_indicator)
-        {
-          case 0:
-            return inner_composition;
-          case 1:
-            return outer_composition;
-          default:
-            Assert (false, ExcMessage ("Unknown boundary indicator. The given boundary indicator should be 0 or 1."));
-            return std::numeric_limits<double>::quiet_NaN();
-        }
+      const std::string boundary_name = geometry_model.translate_id_to_symbol_name(boundary_indicator);
+
+      double X_out=NAN;
+      if (boundary_name == "inner")
+        X_out=inner_composition;
+      if (boundary_name =="outer")
+        X_out=outer_composition;
+
+      return X_out;
```

Comment:
Why don't you do it exactly as it was done for the spherical shell, so that you still have the Assert() in case a different boundary indicator than 'inner' or 'outer' is passed to the function?
Then you also don't need the X_out any more. And can you add spaces between == and 'outer'?